pyramid_swagger/api.py

```diff
@@ -266,23 +266,49 @@ def _get_target_url(spec, target, current_path=''):
         )
 
 
+def _ensure_sane_key(fragment, mar_target):
+    '''
+    This function patches a nasty bug with the
+    'allOff' keyword in a spec, our resolved
+    ref keys become corrupted.
+
+    explanation:
+        sometimes the fragment becomes:
+        '/definitions/:........swagger|..definitions..A'
+        and which makes the marshaled_target:
+        ':........swagger|..definitions..:........swagger|..definitions..A'
```

Comment:
Isn't it a bug that we're generating these doubly nested paths? Can we fix that directly in the target generation logic instead of patching it after the fact in this function?